libfreerdp/codec/progressive.c

```diff
@@ -357,7 +357,7 @@ static INLINE BOOL progressive_set_surface_data(PROGRESSIVE_CONTEXT* progressive
 	key = ((ULONG_PTR)surfaceId) + 1;
 
 	if (pData)
-		return HashTable_Add(progressive->SurfaceContexts, (void*)key, pData) >= 0;
+		return HashTable_Insert(progressive->SurfaceContexts, (void*)key, pData);
```

Comment:
Not your code, but the indexing of surfaces shall be changed to be a pointer on surfaceId and not the surfaceId itself, to remove the +1 trick for non null keys.